groups/ntc/ntco/ntco_iocp.cpp

```diff
@@ -2556,18 +2466,19 @@ ntsa::Error Iocp::detachSocketAsync(
         }
     }
 
-    if (NTCCFG_LIKELY(socketContext->trySetDetachRequired())) {
-        if (socketContext->trySetDetachScheduled()) {
-            socket->setProactorContext(bsl::shared_ptr<void>());
-
-            this->execute(NTCCFG_BIND(&ntcs::Dispatch::announceDetached,
-                                      socket,
-                                      socket->strand()));
+    error = context->detach();
+    if (error) {
+        if (error == ntsa::Error(ntsa::Error::e_WOULD_BLOCK)) {
+            return ntsa::Error();
         }
+        return error;
     }
-    else {
-        return ntsa::Error::invalid();
-    }
+
+    socket->setProactorContext(bsl::shared_ptr<void>());
```

Comment:
Do you agree that we still need to ensure synchronization between `setProactorContext` and `getProactorContext`?

Reply:
I don't have an immediate answer for how to fix this. Let's discuss possible solutions going forward.